micrometer-meter-provider/src/main/java/io/opentelemetry/contrib/metrics/micrometer/internal/instruments/MicrometerDoubleCounter.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.metrics.micrometer.internal.instruments;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.DoubleCounter;
+import io.opentelemetry.api.metrics.DoubleCounterBuilder;
+import io.opentelemetry.api.metrics.ObservableDoubleCounter;
+import io.opentelemetry.api.metrics.ObservableDoubleMeasurement;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.contrib.metrics.micrometer.internal.state.InstrumentState;
+import io.opentelemetry.contrib.metrics.micrometer.internal.state.MeterSharedState;
+import java.util.function.Consumer;
+import javax.annotation.Nullable;
+
+final class MicrometerDoubleCounter extends AbstractCounter implements DoubleCounter {
+
+  private MicrometerDoubleCounter(InstrumentState instrumentState) {
+    super(instrumentState);
+  }
+
+  @Override
+  public void add(double value) {
+    counter(Attributes.empty()).increment(value);
+  }
+
+  @Override
+  public void add(double value, Attributes attributes) {
+    counter(attributes).increment(value);
+  }
+
+  @Override
+  public void add(double value, Attributes attributes, Context context) {
+    counter(attributes).increment(value);
+  }
+
+  public static DoubleCounterBuilder builder(
+      MeterSharedState meterSharedState,
+      String name,
+      @Nullable String description,
+      @Nullable String unit) {
+    return new Builder(meterSharedState, name, description, unit);
+  }
+
+  private static class Builder extends AbstractInstrumentBuilder<Builder>
+      implements DoubleCounterBuilder {
+    private Builder(
+        MeterSharedState meterSharedState,
+        String name,
+        @Nullable String description,
+        @Nullable String unit) {
+      super(meterSharedState, name, description, unit);
+    }
+
+    @Override
+    public Builder self() {
+      return this;
+    }
+
+    @Override
+    public MicrometerDoubleCounter build() {
+      return new MicrometerDoubleCounter(createInstrumentState());
+    }
+
+    @Override
+    public ObservableDoubleCounter buildWithCallback(
+        Consumer<ObservableDoubleMeasurement> callback) {
+      MicrometerDoubleCounter instrument = build();
+      return instrument.registerDoubleCallback(
+          callback,
+          new ObservableDoubleMeasurement() {
+            @Override
+            public void record(double value) {
+              instrument.counter(Attributes.empty()).increment(value);
```

Comment:
All async OTel instruments report absolute values, not deltas; see [the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#asynchronous-counter-creation)

> Note: Unlike [Counter.Add()](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#add) which takes the increment/delta value, the callback function reports the absolute value of the counter. To determine the reported rate the counter is changing, the difference between successive measurements is used.

Maybe instead of using a micrometer `Counter`, a `FunctionCounter` would be better suited here?

(this comment applies to all async instruments here, not just the DoubleCounter)

Reply:
That is interesting, and makes a lot more sense.  I'll revisit these instruments.